android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java

```diff
@@ -269,35 +251,26 @@ public UserSessionInfo getUserSessionInfo() {
         // TODO: a way to distinguish if session is null because we haven't signed on, or if it
         // was invalidated
         UserSessionInfo session = apiClientProvider.getUserSessionInfoProvider()
-                .retrieveCachedSession(getSignInFromStore());
+                .retrieveCachedSession(accountDAO.getSignIn());
 
         if (session != null) {
-            dao.setUserSessionInfo(session);
+            accountDAO.setUserSessionInfo(session);
             return session;
         }
 
-        return dao.getUserSessionInfo();
-    }
-
-    private SignIn getSignInFromStore() {
-        String email = dao.getEmail();
-        String password = dao.getPassword();
-
-        return new SignIn()
-                .study(config.getStudyId())
-                .email(email)
-                .password(password);
+        return accountDAO.getUserSessionInfo();
     }
 
-    class AuthManagerComponentCallback implements ComponentCallbacks {
-        @Override
-        public void onConfigurationChanged(Configuration newConfig) {
-            initApiClientProvider();
-        }
-
-        @Override
-        public void onLowMemory() {
-
-        }
+    /**
+     * Call Bridge to get session. Cached session is updated as a side-effect.
+     *
+     * @return Bridge session
+     */
+    @NonNull
+    public Single<UserSessionInfo> getLatestUserSessionInfo() {
```

Comment:
I made a comment earlier about wanting to provide a sync and async method. The async one here would be when you are looking for a new Session, not from cache. 

However, the cached Session should already remain up-to-date with respect to overall consent state (which seems to be most important), since a 412 will also cause the session in the cache to get updated. 

I'll look into making it protected, there might not be a need after all